File: 프로그래머스/두 개 뽑아서 더하기/plus.cpp
#include <string>
#include <vector>
#include <set>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    sort(numbers.begin(), numbers.end());
    set<int> s;
    for (int i = 0; i < numbers.size() - 1; i++) {
        for (int j = i + 1; j < numbers.size(); j++) {
            int tempnum = numbers[i] + numbers[j];
            if (s.find(tempnum) == s.end()) s.insert(tempnum);
        }
    }
    for (auto i = s.begin(); i != s.end(); i++) {
        answer.push_back(*i);
    }
    sort(answer.begin(), answer.end());
    return answer;
}

File: 프로그래머스/두 개 뽑아서 더하기/plus.java
import java.util.*;
class Solution {
    public int[] solution(int[] numbers) {
        HashSet<Integer> set = new HashSet<Integer>();
        for(int i=0; i<numbers.length-1;i++){
            for(int j=0; j<numbers.length;j++){
                if(i==j) continue;
                int curnum = numbers[i]+numbers[j];
                set.add(curnum);
            }
        }
        Integer[] arr = new Integer[set.size()];
        set.toArray(arr);
        int answer[] = Arrays.stream(arr).mapToInt(Integer::intValue).toArray();
        Arrays.sort(answer);
        return answer;
    }
}
public class test{
    public static void main(String[] args){

    }
}
